implementations/rust/ockam/ockam_vault/src/signer_impl.rs

```diff
@@ -22,13 +26,23 @@ impl Signer for SoftwareVault {
                         CURVE25519_SECRET_LENGTH
                     ))
                     .sign(data.as_ref(), &nonce);
-                    Ok(sig)
+                    Ok(sig.into())
                 } else {
                     Err(VaultError::InvalidKeyType.into())
                 }
             }
             SecretType::Bls => {
-                Ok([0u8; 64]) // FIXME
+                if key.len() == 32 {
+                    let bls_secret_key = SecretKey::from_bytes(array_ref!(key, 0, 32)).unwrap();
```

Comment:
@feniks65 @mikelodder7 is this the proper way of using BLS for simple signing? Could we use ECDSA? Would it be simpler/more effective?

Reply:
@SanjoDeundiak @mikelodder7 
I have used an example from the following file implementations/rust/ockam/signature_bbs_plus/README.md
thinking that this is how I should proceed with BBS+ signature signing.

Reply:
@feniks65 I think BBS+ is a little bit overkill in this use case, since we don't use its full potential, but since this is the only thing we have currently implemented for BLS, let's merge it and revisit the decision in the future. Nice job!